File: seminar_5/task_5/Khamitov.Islam/task_5.py
# Задача 5 
# Максимальная выгода . Есть массив чисел ,каждое - акция. Как наиболее выгодно купить и продать?

def maxProfit(prices):
    profit = 0
    min_price = prices[0]
    
    for cur_pr_index in range(len(prices)):
        profit = max(profit, prices[cur_pr_index] - min_price)
        min_price = min(prices[cur_pr_index], min_price)
        
    return profit

# example
prices = [2,15 ,14,3,6,9,48,15,1,2,12]
print(maxProfit(prices))

File: seminar_5/task_5/churilkin.artem/task5.cpp
#include <vector>
#include <algorithm>

int maxProfit(std::vector<int> prices){
    int profit = 0;
    int min_price = prices[0];
    for( int currentPriceIndex = 1; currentPriceIndex < prices.size(); currentPriceIndex++){
        profit = std::max(profit, prices[currentPriceIndex] - min_price);
        min_price = std::min (prices[currentPriceIndex], min_price);
    }

    return profit;

}

int main(){
    return 0;
}